libpod/container_internal.go

```diff
@@ -1307,16 +1311,35 @@ func (c *Container) stop(timeout uint) error {
 		}
 	}
 
-	// Set the container state to "stopping" and unlock the container
-	// before handing it over to conmon to unblock other commands.  #8501
-	// demonstrates nicely that a high stop timeout will block even simple
-	// commands such as `podman ps` from progressing if the container lock
-	// is held when busy-waiting for the container to be stopped.
+	// OK, the following code looks a bit weird but we have to make sure we can stop
+	// containers with the restart policy always, to do this we have to set
+	// StoppedByUser even when there is nothing to stop right now. This is due the
```

Comment:
As long as we're repushing for language: `s/due the/due to the/`. Also, the "then restarts it" in the line below, the "contianer" one, should maybe be "restarting it"?